src/utils/shell.ts

```diff
@@ -0,0 +1,150 @@
+import cluster from 'cluster';
+import { resolve } from 'path';
+import * as shell from 'shelljs' ;
+import * as env from './environment';
+import * as vscode from 'vscode';
+
+shell.config.execPath = String(shell.which('node'));
+
+// execute a command and return a promise of the output as string
+export function executeShellCommand(command: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+		let execution = shell.exec(command);
+		if (execution.code === 0) {
+			resolve(execution.stdout);
+		}
+		reject(execution.stderr);
+	});
+}
+
+// check if a cli tool exists and return a promise
+export  function checkToolExists(tool: string): Promise<void> {
+	return new Promise((resolve, reject) => {
+		let execution = shell.exec(`command -v ${tool}`);
+		if (execution.code === 0) {
+			resolve();
+		}
+		reject();
+	});
+}
+
+export async function sleep(ms: number): Promise<void> {
+	return new Promise((resolve, _reject) => setTimeout(() => resolve(), ms));
+}
+
+
+// init kind clusters 
+function initClusters(clusters: Array<env.Cluster>): Promise<string> {
+
+	console.log('creating clusters');
```

Comment:
who are we writing this to the console for?

if it's meant for the user - let's put it in the progress windows.
if it's meant for developers, do we really need it, and if so shouldn't this be "debug" instead of "log"?
🤔 

This goes for all the rest of the _console.log_ statements in this file.

Reply:
you are right